### PR TITLE
SecurityReviewScore can consider changes in projects

### LINKED ISSUE #586: SecurityReviewScore can consider changes in the project
The older review is, the more likely it is that new changes have been added. Those new changes were not covered by the review. Plus, new vulnerability types and attacks may be discovered since the review.

In general, it would be good to update the scoring function to take into account the changes in the project after review. The less changes, the less impact on the score. For example, we can use commit history.

_Originally posted by @artem-smotrakov in https://github.com/SAP/fosstars-rating-core/pull/577#discussion_r649890682_

### PR DESCRIPTION
**Pull request by artem-smotrakov**
Here are main updates:

- Updated `SecurityReview` to hold info about amount of changes in the project since the review was done.
- Updated `SecurityReviewsFromOpenSSF` provider to calculate amount of changes in the project since a specific date.
- Updated `SecurityReviewScore` to take into account amount of changes in the project since the review was done.

Closes #586

### CODE REVIEW

File: src/main/java/com/sap/oss/phosphor/fosstars/data/github/LocalRepository.java [hunk 1]
```diff
@@ -334,6 +341,79 @@ public void close() {
     repository.close();
   }
 
+  /**
+   * Estimate how much the project has changed since a specified data.
```
Comment (sourabhsparkala) [inline src/main/java/com/sap/oss/phosphor/fosstars/data/github/LocalRepository.java:345]: Should data -> date?

File: src/main/java/com/sap/oss/phosphor/fosstars/data/github/LocalRepository.java [hunk 2]
```diff
@@ -334,6 +341,79 @@ public void close() {
...
+   *
+   * @param date The date.
+   * @param toConsider Defines which files should be considered.
+   * @return An estimated amount of changes in the project.
+   * @throws IOException If something went wrong.
+   */
+  public Double changedSince(Date date, Predicate<Path> toConsider) throws IOException {
+    loadCommitsIfNecessary();
+
+    GitCommit head = commits.get(0);
```
Comment (sourabhsparkala) [inline src/main/java/com/sap/oss/phosphor/fosstars/data/github/LocalRepository.java:358]: If commits are empty even after trying to load commits and there were no exceptions but still empty. Could this happen?

File: src/test/java/com/sap/oss/phosphor/fosstars/data/github/SecurityReviewsFromOpenSSFTest.java [hunk 3]
```diff
@@ -151,17 +162,53 @@ public void testReadReviewDateFrom() throws JsonProcessingException {
...
-    Path repositoryDirectory
+  public void testFetchValueFor() throws IOException, GitAPIException, ParseException {
+    Path securityReviewDirectory
         = Files.createTempDirectory(SecurityReviewsFromOpenSSFTest.class.getName());
-    try {
-      LocalRepository repository = mock(LocalRepository.class);
-      TestGitHubDataFetcher.addForTesting(SECURITY_REVIEWS_PROJECT, repository);
+    Path directory = Files.createTempDirectory(getClass().getName());
+    try (Repository repository = FileRepositoryBuilder.create(directory.resolve(".git").toFile());
+        Git git = new Git(repository)) {
```
Comment (sourabhsparkala) [inline src/test/java/com/sap/oss/phosphor/fosstars/data/github/SecurityReviewsFromOpenSSFTest.java:170]: Seems to be an indentation issue

**Review by sourabhsparkala — APPROVED** [review]
Minor comments

File: src/main/java/com/sap/oss/phosphor/fosstars/data/github/LocalRepository.java (hunk 2 above)
Comment (artem-smotrakov) [reply src/main/java/com/sap/oss/phosphor/fosstars/data/github/LocalRepository.java:358]: It is better to add a check here.

File: src/test/java/com/sap/oss/phosphor/fosstars/data/github/SecurityReviewsFromOpenSSFTest.java (hunk 3 above)
Comment (artem-smotrakov) [reply src/test/java/com/sap/oss/phosphor/fosstars/data/github/SecurityReviewsFromOpenSSFTest.java:170]: That was auto-formatted according to the formatting rules. I can add an extra space here.

**Pull request merged**